sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/util/GcsUtil.java

```diff
@@ -871,6 +876,10 @@ private void rewriteHelper(
     for (int i = 0; i < srcList.size(); i++) {
       final GcsPath sourcePath = GcsPath.fromUri(srcList.get(i));
       final GcsPath destPath = GcsPath.fromUri(destList.get(i));
+      if (ignoreExistingDest && !sourcePath.getBucket().equals(destPath.getBucket())) {
```

Comment:
Unless we actually plumb this through to the user as a PipelineOption this exception will not be actionable to the end user.

Won't this end up being a non-actionable hard regression for some users since we always set this at FileBasedSink ?

https://github.com/apache/beam/blob/a0fbe00ef12b72ec89672ab32ccc6d6331ca5edd/sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileBasedSink.java#L777

Reply:
This was relying on FileBasedSink using Filesystems.rename which handles UnsupportedOperationExceptions in underlying filel system rename by falling back to matching to filter files.

https://github.com/apache/beam/blob/a0fbe00ef12b72ec89672ab32ccc6d6331ca5edd/sdks/java/core/src/main/java/org/apache/beam/sdk/io/FileSystems.java#L310